sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReceiveMessageAndSettleSample.java

```diff
@@ -0,0 +1,100 @@
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sample demonstrates how to receive an {@link ServiceBusReceivedMessage} from an Azure Service Bus Queue and settle
+ * it. Settling of message include accept, defer and abandon the message as needed.
+ */
+public class ReceiveMessageAndSettleSample {
+    /**
+     * Main method to invoke this demo on how to receive an {@link ServiceBusMessage} from an Azure Service Bus
+     * Queue
+     *
+     * @param args Unused arguments to the program.
+     */
+    public static void main(String[] args) {
+
+        // The connection string value can be obtained by:
+        // 1. Going to your Service Bus namespace in Azure Portal.
+        // 2. Go to "Shared access policies"
+        // 3. Copy the connection string for the "RootManageSharedAccessKey" policy.
+        String connectionString = System.getenv("AZURE_SERVICEBUS_CONNECTION_STRING");
+
+        // Create a receiver.
+        // "<<fully-qualified-namespace>>" will look similar to "{your-namespace}.servicebus.windows.net"
+        // "<<queue-name>>" will be the name of the Service Bus queue instance you created
+        // inside the Service Bus namespace.
+
+        ServiceBusReceiverAsyncClient receiverAsyncClient = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .buildReceiverClientBuilder()
+            .receiveMode(ReceiveMode.PEEK_LOCK)
+            .isLockAutoRenewed(true)
+            .queueName("<<queue-name>>")
+            .maxAutoLockRenewalDuration(Duration.ofSeconds(2))
+            .buildAsyncClient();
+
+        Disposable subscription = receiverAsyncClient.receive()
+            .subscribe(received -> {
+                Instant latest = Instant.MIN;
+
+                // Simulate some sort of long processing. Sleep should not be used in production system.
+                for (int i = 0; i < 3; i++) {
+                    try {
+                        TimeUnit.SECONDS.sleep(15);
+                    } catch (InterruptedException error) {
+                        System.out.println("Error occurred while sleeping: " + error);
+                    }
+                    latest = received.getLockedUntil();
+                    System.out.println("Message Locked Until " + latest);
+                }
+
+                //This is application business logic to take action based on some application logic.
+                // For demo purpose we are using a property for application logic.
+                String propertyValue = "UNDEFINED";
+                if (received.getProperties().get("SOME_USER_PROPERTY") != null) {
+                    propertyValue = (String) received.getProperties().get("SOME_USER_PROPERTY");
+                }
+
+                switch (propertyValue) {
+                    case "VALUE-1":
+                        System.out.println("Completing message.");
+                        receiverAsyncClient.complete(received).block(Duration.ofSeconds(15));
```

Comment:
* You need to set "isAutoComplete" to false when building this client. Or it'll try to complete or abandon the message again. 
* "Duration.ofSeconds()" is used multiple times and can be a final variable.